Source/Csla.AspNetCore/Blazor/ApplicationContextManagerBlazor.cs

```diff
@@ -65,40 +65,29 @@ public ApplicationContextManagerBlazor(IHttpContextAccessor hca, AuthenticationS
 
     private async Task InitializeUser()
     {
-      if (ActiveCircuitState.CircuitExists)
+      if (HttpContext != null)
+      {
+        var user = HttpContext.User;
+        if (user != null)
+        {
+          CurrentPrincipal = user;
+          SetHostPrincipal(Task.FromResult(new AuthenticationState(user)));
+        }
+      }
+      else
       {
         Task<AuthenticationState> task;
         try
         {
           task = AuthenticationStateProvider.GetAuthenticationStateAsync();
           await task;
         }
-        catch (InvalidOperationException ex)
+        catch (InvalidOperationException)
         {
           task = Task.FromResult(new AuthenticationState(UnauthenticatedPrincipal));
```

Comment:
Isn't the `SetHostPrinciapl(task)` now missing in this case and we end up with no user?

Reply:
I don't think it is our job to set the current principal. Blazor 8 appears to keep HttpContext.User and the AuthenticationStateProvider in sync all by itself.

I really think CSLA was overcomplicating some of this stuff earlier.